File: lintcode/172_remove-element/remove-element.py
# coding:utf-8
'''
@Copyright:LintCode
@Author:   chenweikang
@Problem:  http://www.lintcode.com/problem/remove-element
@Language: Python
@Datetime: 16-04-25 15:37
'''

class Solution:
    """
    @param A: A list of integers
    @param elem: An integer
    @return: The new length after remove
    """
    def removeElement(self, A, elem):
        j = 0
        for i in range(len(A)):
            if A[i] != elem:
                A[j], A[i] = A[i], A[j]
                j += 1
        return j
            

File: lintcode/172_remove-element/remove-element.cpp
/*
@Copyright:LintCode
@Author:   chenweikang
@Problem:  http://www.lintcode.com/problem/remove-element
@Language: C++
@Datetime: 16-04-25 15:34
*/

class Solution {
public:
    /** 
     *@param A: A list of integers
     *@param elem: An integer
     *@return: The new length after remove
     */
    int removeElement(vector<int> &A, int elem) {
        const int n = A.size();
        int j = 0;   
        for (int i = 0; i < n; ++i) {
            if(A[i] != elem) swap(A[j++], A[i]);
        }
        return j;
    }
};